modules/phase_field/src/action/GrandPotentialKernelAction.C

```diff
@@ -244,50 +262,136 @@ GrandPotentialKernelAction::act()
-    for (unsigned int j = 0; j < n_etas + n_grs; ++j)
+    // Chemical Potentials
+    for (unsigned int i = 0; i < n_w; ++i)
     {
-      notarealvector[0] = all_etas[j];
-      params = _factory.getValidParams("CoupledSwitchingTimeDerivative");
+      // SusceptibilityTimeDerivative
+      InputParameters params = _factory.getValidParams("SusceptibilityTimeDerivative");
+      params.set<NonlinearVariableName>("variable") = w_names[i];
+      params.set<MaterialPropertyName>("f_name") = chis[i];
+      params.set<std::vector<VariableName>>("args") = v0;
+      params.set<bool>("implicit") = implicity;
+      params.set<bool>("use_displaced_mesh") = displaced_mesh;
+      kernel_name = "ChiDt_" + w_names[i];
+      _problem->addKernel("SusceptibilityTimeDerivative", kernel_name, params);
+     
+      // MatDiffusion
+      params = _factory.getValidParams("MatDiffusion");
       params.set<NonlinearVariableName>("variable") = w_names[i];
-      params.set<std::vector<VariableName>>("v") = notarealvector;
-      params.set<std::vector<VariableName>>("coupled_variables") = v0;
-      params.set<std::vector<MaterialPropertyName>>("Fj_names") = fj_temp;
-      params.set<std::vector<MaterialPropertyName>>("hj_names") = hj;
       params.set<bool>("implicit") = implicity;
       params.set<bool>("use_displaced_mesh") = displaced_mesh;
-      kernel_name = "Coupled_" + w_names[i] + "_" + all_etas[j];
-      _problem->addKernel("CoupledSwitchingTimeDerivative", kernel_name, params);
-    }
-  }
+      params.set<MaterialPropertyName>("diffusivity") = M[i];
+      kernel_name = "MatDif_" + w_names[i];
+      if (aniso.get(i))
+        _problem->addKernel("MatAnisoDiffusion", kernel_name, params);
+      else
+      {
+        params.set<std::vector<VariableName>>("args") = v0;
+        _problem->addKernel("MatDiffusion", kernel_name, params);
+      }
+     
+      // CoupledSwitchingTimeDerivative
+      for (unsigned int j = 0; j < n_hj; ++j)
+        fj_temp[j] = Fj_w[i * n_hj + j];
+      for (unsigned int j = 0; j < n_etas + n_grs; ++j)
+      {
+        notarealvector[0] = all_etas[j];
+        params = _factory.getValidParams("CoupledSwitchingTimeDerivative");
+        params.set<NonlinearVariableName>("variable") = w_names[i];
+        params.set<std::vector<VariableName>>("v") = notarealvector;
+        params.set<std::vector<VariableName>>("args") = v0;
+        params.set<std::vector<MaterialPropertyName>>("Fj_names") = fj_temp;
+        params.set<std::vector<MaterialPropertyName>>("hj_names") = hj;
+        params.set<bool>("implicit") = implicity;
+        params.set<bool>("use_displaced_mesh") = displaced_mesh;
+        kernel_name = "Coupled_" + w_names[i] + "_" + all_etas[j];
+        _problem->addKernel("CoupledSwitchingTimeDerivative", kernel_name, params);
+      } //     for (unsigned int j = 0; j < n_etas + n_grs; ++j) coupledSwitchingTimeDeriv
+    } //for (unsigned int i = 0; i < n_w; ++i) chempot
```

Comment:
It looks like there are some misaligned braces in a few places. This can be fixed by running the Clang-format utility on all your modified files. There are extensions to most text editors, such as VSCode or Atom, that will allow you to run this within the editor. The MOOSE test suite will also check for conformance with clang-format as part of the test suite and throw and error if it does not comply, it will give you the option to apply a patch to fix it, but it's easier just to install a plugin to your text editor and run prior to submitting

Reply:
And please remove the commented out loop headers. Indentation should make this clear enough.

Reply:
While you're at it, please also format the input files using the MOOSE VSCode/Atom plugin (or the `hit format file.i` command.